app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java

```diff
@@ -120,12 +120,12 @@ private PictureUtil getPictureUtility() {
         .thenCompose(
             (aVoid) -> {
               // update user info
-              UserUtil.getSingleInstance()
+              DependencyFactory.getCurrentUserRepository()
```

Comment:
should make the user repository a local field instead?

Reply:
Done